docs/Reference/Server.md

```diff
@@ -1640,6 +1641,45 @@ const fastify = require('fastify')({
 The handler is bound to the Fastify instance and is fully encapsulated, so
 different plugins can set different logger factories.
 
+#### setGenReqId
+<a id="set-gen-req-id"></a>
+
+`fastify.setGenReqId(function (rawReq))` Function for setting the request-id for
```

Comment:
It might be worth it to state that we expect a `synchronous` function, so users are aware of that